### ISSUE TITLE
IndexOutOfRangeException when using dprep with nimbusml

### ISSUE THREAD

**Issue by vinodshanbhag** [microsoft/NimbusML#268]
We have run into another peculiar issue with nimbusml and dprep combo.

If we run the following script, then filedatastream based training succeedes but dataprep based training failed with IndexOutOfRange exception.

However if you set caching=’Auto’ both of them succeed.

Unclear where the bug belongs – dprep side or nimbus side.

from nimbusml import FileDataStream, Pipeline
import nimbusml.linear_model as nml_linear
from nimbusml.feature_extraction.categorical import OneHotVectorizer
from nimbusml.preprocessing.missing_values import Handler
import azureml.dataprep as dprep
from nimbusml import DprepDataStream

X_train = FileDataStream.read_csv("C:\\benchmarkigndatasets\\NYCTaxiTipPrediction__train.csv")
X_test = FileDataStream.read_csv("C:\\benchmarkigndatasets\\NYCTaxiTipPrediction__valid.csv")

X_train_dprep = dprep.auto_read_file("C:\\benchmarkigndatasets\\NYCTaxiTipPrediction__train.csv")
X_test_dprep = dprep.auto_read_file("C:\\benchmarkigndatasets\\NYCTaxiTipPrediction__valid.csv")

pipe = Pipeline([
OneHotVectorizer(columns={'vendor_id': 'vendor_id', 'payment_type': 'payment_type', 'passenger_count': 'passenger_count','rate_code': 'rate_code'},
                 max_num_terms=1000000,
                 output_kind='Indicator',
                 sort='ByOccurrence',
                 term=None),
Handler(columns={'trip_distance': 'trip_distance', 'trip_time_in_secs': 'trip_time_in_secs'},
    concat=False,
        impute_by_slot=True,
        replace_with='Mean'),

    nml_linear.FastLinearRegressor(
        feature=['vendor_id', 'payment_type', 'passenger_count', 'rate_code', 'trip_distance', 'trip_time_in_secs'],
        label='fare_amount',
        l2_regularization=None,
        l1_threshold=None,
        convergence_tolerance=0.01,
        maximum_number_of_iterations=None,
        shuffle=True,
        bias_learning_rate=1,
        caching='None')
])

try:
    pipe.fit(X_train)
    metrics, scores = pipe.test(X_test)
    print(metrics)
except Exception as e:
    print('FileDataStream based training failed')
    print(e)

try:
    pipe.fit(DprepDataStream(X_train_dprep))
    metrics, scores = pipe.test(DprepDataStream(X_test_dprep))
    print(metrics)
except Exception as e:
    print('DprepDataStream based training failed')
    print(e)

**Comment by ganik** [comment]
this has been fixed in dprep
